src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -482,7 +485,22 @@ static inline void uct_dc_mlx5_iface_dci_free(uct_dc_mlx5_iface_t *iface,
         return;
     }
 
-    uct_dc_mlx5_iface_dci_release(iface, dci_index);
+    if (!sched) {
```

Comment:
IMO it should be simpler:
- no need to have release list or elements
- progress just goes over all dci in the iface  and releases those without outstanding/pending
- this way no need changes in dci_free (which increase code size of inlined function)

Reply:
ok, done